### PR TITLE
Changes to options struct with handlers patch

### PR DESCRIPTION
**Pull request by joeydlee95**
-take care to look at the fix I made to addHandler
-Also working on tests, currently there is a seg fault in webserver test (Probably fixed after Evan's fix)
- so far the options test is 73% coverage will attempt to increase once StaticHandler and EchoHandler works
- our static module is at 82% and lowest is webserver 60% LOL

### CODE REVIEW

File: tests/options_test.cc [hunk 1]
```diff
@@ -59,47 +52,27 @@ TEST(OptionsLoadFileTest, ASimpleFile) {
...
+	std::stringstream mod("port 8080; path /echo {}");
 
 	Options opt;
 	EXPECT_FALSE(opt.loadOptionsFromStream(&mod));
 }
 
-TEST(OptionsLoadStreamModTest, NotValidType) {
-	std::stringstream mod("server { module { type = shouldfail; } }");
+TEST(OptionsLoadStreamHandTest, NotValidType) {
+	std::stringstream mod("port 8080; module EchoHandler {}");
```
Comment (evangrayk) [inline tests/options_test.cc]: This test passes for the wrong reason I think. Maybe change it to something like 
`path / NotARealHandlerType { }`

**Review by evangrayk — COMMENTED** [review]
Looks great! The handler merge is totally correct. One thing is that I had assumed the old format in one other place: in request_handler.cc I check if the type string is `echo` not `EchoHandler`. I know it's not exactly part of the config but I think changing that might fit well in this PR. Currently I think it would only work correctly if your config looked like 
`path / echo {}` instead of `path / EchoModule {}` which is what we intended.

**Comment by evangrayk** [comment]
Also, RE: coverage, it'll definitely go up once we have other handlers implemented. I think some things we really can't cover in tests, like strange failures and if boost is working properly. So I think as long as we're 90-100% in places where we aren't relying on boost's functionality, I think that's good.

**Review by Filisha — APPROVED** [review]
I think some of these should have been separate pull requests to aid the reviewer in understanding different functionalities.
Looks good otherwise.

**Comment by evangrayk** [comment]
@Filisha I kinda think it is one sort of logical chunk of changes, so it makes sense as one PR. It's changing from the old config file format to the new one. It got to be a little big because it has a few changes, but you can always go back and look at each individual commit if it helps!

**Pull request merged**